prometheus/promhttp/delegator.go

```diff
@@ -86,9 +88,7 @@ func (d closeNotifierDelegator) CloseNotify() <-chan bool {
 func (d flusherDelegator) Flush() {
 	// If applicable, call WriteHeader here so that observeWriteHeader is
```

Comment:
The "If applicable" is now a bit confusing (as there is no `if` statement anymore). How about changing this and the other comments to "Call WriteHeader here so that observeWriteHeader is handled appropriately if still needed."